clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1637,6 +1637,18 @@ static std::string eraseAnonNamespace(std::string S) {
   return S;
 }
 
+static bool checkEnumTemplateParameter(const EnumDecl *ED,
```

Comment:
Check functions typically return 'true' on an error.  Also allows return Diag.Report (I think?)

Reply:
Alright. Will change that.